src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/sge/SGEJobManager.groovy

```diff
@@ -0,0 +1,143 @@
+import de.dkfz.roddy.execution.jobs.BEJob
+import de.dkfz.roddy.execution.jobs.GenericJobInfo
+import de.dkfz.roddy.execution.jobs.JobManagerCreationParameters
+import de.dkfz.roddy.execution.jobs.ProcessingCommands
+import de.dkfz.roddy.execution.jobs.cluster.pbs.PBSJobManager
+import de.dkfz.roddy.execution.jobs.cluster.pbs.PBSResourceProcessingCommand
+import de.dkfz.roddy.tools.BufferUnit
+import groovy.transform.CompileStatic
+import sun.reflect.generics.reflectiveObjects.NotImplementedException
+
+/**
+ * Created by michael on 20.05.14.
+ */
+@CompileStatic
+class SGEJobManager extends PBSJobManager {
+    SGEJobManager(BEExecutionService executionService, JobManagerCreationParameters parms) {
+        super(executionService, parms)
+    }
+
+    @Override
+    SGECommand createCommand(GenericJobInfo jobInfo) {
+        throw new NotImplementedException()
+    }
+
+    SGECommand createCommand(BEJob job, List<ProcessingCommands> processingCommands, String command, Map<String, String> parameters, Map<String, Object> tags, List<String> dependencies, File logDirectory) {
+        SGECommand sgeCommand = new SGECommand(this, job, job.jobID.toString(), processingCommands, parameters, tags, null, dependencies, command, logDirectory)
+        return sgeCommand
+    }
+
+    @Override
+    SGECommand createCommand(BEJob job, String jobName, List<ProcessingCommands> processingCommands, File tool, Map<String, String> parameters, List<String> dependencies) {
+        throw new NotImplementedException()
+    }
+
+    SGECommand createCommand(BEJob job) {
+        return new SGECommand(this, job, job.jobName, [], job.parameters, [:], [], job.dependencyIDsAsString, job.tool?.getAbsolutePath() ?: job.getToolScript(), job.getLoggingDirectory())
+    }
+//    @Override
+//    public void addSpecificSettingsToConfiguration(Configuration configuration) {
+//        configuration.getConfigurationValues().add(new ConfigurationValue("RODDY_JOBID", "${JOB_ID-}"));
+//        configuration.getConfigurationValues().add(new ConfigurationValue("RODDY_SCRATCH", "/tmp/roddyScratch/${JOB_ID}"));
+//        configuration.getConfigurationValues().add(new ConfigurationValue("RODDY_AUTOCLEANUP_SCRATCH", "true"));
+//    }
+
+    @Override
+    ProcessingCommands parseProcessingCommands(String processingString) {
+        return convertPBSResourceOptionsString(processingString)
+    }
+
+    @Override
+    String getResourceOptionsPrefix() {
+        return "SGEResourceOptions_"
+    }
+
+    @Override
+    ProcessingCommands convertResourceSet(BEJob job, ResourceSet resourceSet) {
+        StringBuilder sb = new StringBuilder()
+        sb.append(" -V") //TODO Think if default SGE options should go somewhere else?
+        if (resourceSet.isMemSet()) {
+            String memoryInMB = resourceSet.getMem().toString(BufferUnit.M)
+            sb << " -l " << "s_data" << "=" << memoryInMB
+        }
+        if (resourceSet.isStorageSet()) {
+        }
```

Comment:
Empty if statement!

Reply:
Is this actually supported by any job manager? LSF? If not, we may consider removing this ballast.

Reply:
Yes it is supported by PBS. But in our local installation the system had a bug (old PBS!) That is, why we never really used it.

If we unify this code, I'd definitely make it possible to enable or disable these options and also to set a default value in the job manager.